apps/dashboard/src/components/primitives/control-input/variable-plugin/variable-pill-widget.ts

```diff
@@ -84,42 +107,140 @@ export class VariablePillWidget extends WidgetType {
   toDOM() {
     const span = document.createElement('span');
     const content = document.createElement('span');
-    content.textContent = this.variableName;
+    content.textContent = this.getDisplayVariableName();
     const before = document.createElement('span');
 
     const pillStyles = this.createPillStyles();
     Object.assign(span.style, pillStyles);
 
     const beforeStyles = this.createBeforeStyles();
     Object.assign(before.style, beforeStyles);
+    const hasIssues = this.getVariableIssues().length > 0;
+
+    if (hasIssues) {
+      before.style.color = 'hsl(var(--error-base))';
+      before.style.backgroundImage = `url("/images/error-circle-outline.svg")`;
+      before.style.backgroundSize = 'cover';
+    }
 
     const contentStyles = this.createContentStyles();
     Object.assign(content.style, contentStyles);
 
-    // Stores the complete variable expression including any filters
     span.setAttribute('data-variable', this.fullVariableName);
-
     span.setAttribute('data-start', this.start.toString());
     span.setAttribute('data-end', this.end.toString());
-
-    // Contains the clean variable name shown to the user
     span.setAttribute('data-display', this.variableName);
 
     span.appendChild(before);
     span.appendChild(content);
 
-    if (this.hasFilters) {
-      const after = document.createElement('span');
-      const afterStyles = this.createAfterStyles();
-      Object.assign(after.style, afterStyles);
-      span.appendChild(after);
-    }
-
     span.addEventListener('mousedown', this.clickHandler);
 
+    this.renderFilters(span);
+
+    span.addEventListener('mouseenter', () => {
+      if (!this.tooltipElement) {
+        this.tooltipElement = this.renderTooltip(span);
+      }
+
+      if (hasIssues) {
+        span.style.backgroundColor = 'hsl(var(--error-base) / 0.025)';
+      }
+    });
+
+    span.addEventListener('mouseleave', () => {
+      if (this.tooltipElement) {
+        document.body.removeChild(this.tooltipElement);
+        this.tooltipElement = null;
+      }
+
+      span.style.backgroundColor = 'hsl(var(--bg-white))';
+    });
+
     return span;
   }
 
+  renderFilters(parent: HTMLElement) {
+    if (!this.filters?.length) return;
+
+    const firstFilter = this.filters[0];
+    const firstFilterName = firstFilter.split(':')[0];
+    const firstFilterParams = firstFilter.split(':')[1]?.split(',')?.[0];
+    const parsedFilterParams = parseParams(firstFilterParams);
```

Comment:
Please move this parsing logic to `utils/liquid.ts` file. It could be also part of the `parseVariable` function, but we don't always need to run it.